assets/js/blocks/products/utils.js

```diff
@@ -11,14 +11,39 @@ export const getBlockClassName = ( blockClassName, attributes ) => {
 	const { className, contentVisibility } = attributes;
 
 	return classNames( blockClassName, className, {
-		'has-image': contentVisibility.image,
-		'has-title': contentVisibility.title,
-		'has-rating': contentVisibility.rating,
-		'has-price': contentVisibility.price,
-		'has-button': contentVisibility.button,
+		'has-image': contentVisibility && contentVisibility.image,
+		'has-title': contentVisibility && contentVisibility.title,
+		'has-rating': contentVisibility && contentVisibility.rating,
+		'has-price': contentVisibility && contentVisibility.price,
+		'has-button': contentVisibility && contentVisibility.button,
 	} );
 };
 
+/**
+ * Adjust attributes on load to set defaults so default attributes get saved.
+ *
+ * Ref: https://github.com/WordPress/gutenberg/issues/7342
+ *
+ * @param {Object} blockAttributes Original block attributes.
+ * @param {Object} blockType       Block type settings.
+ *
+ * @return {Object} Filtered block attributes.
+ */
+export const setBlockAttributeDefaults = ( blockAttributes, blockType ) => {
+	Object.keys( blockType.attributes ).map( ( key ) => {
```

Comment:
Would be good to only run this on our blocks.  We're using a non-official property on the block config here, there's risk another block somewhere might be using `defaults` in a different way.

This also applies to the other filter implementations in this pull (I won't repeat this comment for `withDefaultAttributes`).

Reply:
Done. Added a check for the block prefix.